util/ranger/detacher.go

```diff
@@ -876,3 +877,402 @@ func MergeDNFItems4Col(ctx sessionctx.Context, dnfItems []expression.Expression)
 	}
 	return mergedDNFItems
 }
+
+// AddGcColumnCond add the `tidb_shard(x) = xxx` to the condition
+// @param[in] cols          the columns of shard index, such as [tidb_shard(a), a, ...]
+// @param[in] accessCond    the conditions relative to the index and arranged by the index column order.
+//                          e.g. the index is uk(tidb_shard(a), a, b) and the where clause is
+//                          `WHERE b = 1 AND a = 2 AND c = 3`, the param accessCond is {a = 2, b = 1} that is
+//                          only relative to uk's columns.
+// @param[in] columnValues  the values of index columns in param accessCond. if accessCond is {a = 2, b = 1},
+//                          columnValues is {2, 1}. if accessCond the "IN" function like `a IN (1, 2)`, columnValues
+//                          is empty.
+// @retval -  []expression.Expression   the new condtions after adding `tidb_shard() = xxx` prefix
+//            error                     if error gernerated, return error
+func AddGcColumnCond(sctx sessionctx.Context,
+	cols []*expression.Column,
+	accessesCond []expression.Expression,
+	columnValues []*valueInfo) ([]expression.Expression, error) {
+
+	if cond := accessesCond[1]; cond != nil {
+		if f, ok := cond.(*expression.ScalarFunction); ok {
+			switch f.FuncName.L {
+			case ast.EQ:
+				return AddGcColumn4EqCond(sctx, cols, accessesCond, columnValues)
+			case ast.In:
+				return AddGcColumn4InCond(sctx, cols, accessesCond)
+			}
+		}
+	}
+
+	return accessesCond, nil
+}
+
+// AddGcColumn4InCond add the `tidb_shard(x) = xxx` for `IN` condition
+// For param explanation, please refer to the function `AddGcColumnCond`.
+// @retval -  []expression.Expression   the new condtions after adding `tidb_shard() = xxx` prefix
+//            error                     if error gernerated, return error
+func AddGcColumn4InCond(sctx sessionctx.Context,
```

Comment:
This func also doesn't need to be exported?

Reply:
For unit test